src/server/campaignd/server.cpp

```diff
@@ -811,14 +812,14 @@ void server::handle_request_campaign(const server::request& req)
 						} else {
 							WRN_CS << "Unable to create an update pack sequence from version (" << from << ") to ("
 								   << to << ") for the addon '" << req.cfg["name"].str() << "'. A full pack will be sent instead!\n";
-							pack_data = config::get_invalid();
+							failed = true;
 							break;
 						}
 					}
 				}
 			}
 
-			if(pack_data && !pack_data.empty()) {
+			if(!failed && pack_data && !pack_data.empty()) {
```

Comment:
It makes no sense  to check `pack_data` here, since `pack_data` is an actual object instead of a reference this is never `false`. (the bool operator of the config class compares the this pointer to the config class, )

Reply:
Very well, removed in https://github.com/wesnoth/wesnoth/pull/5038/commits/44135f3930d7d381eb6c3769d21e2985736861ad .